src/renderer.c

```diff
@@ -50,19 +50,20 @@ typedef struct RenFont {
 } RenFont;
 
 static const char* utf8_to_codepoint(const char *p, unsigned *dst) {
+  const unsigned char *up = (unsigned char*)p;
   unsigned res, n;
   switch (*p & 0xf0) {
-    case 0xf0 :  res = *p & 0x07;  n = 3;  break;
-    case 0xe0 :  res = *p & 0x0f;  n = 2;  break;
+    case 0xf0 :  res = *up & 0x07;  n = 3;  break;
+    case 0xe0 :  res = *up & 0x0f;  n = 2;  break;
     case 0xd0 :
-    case 0xc0 :  res = *p & 0x1f;  n = 1;  break;
-    default   :  res = *p;         n = 0;  break;
+    case 0xc0 :  res = *up & 0x1f;  n = 1;  break;
+    default   :  res = *up;         n = 0;  break;
   }
   while (n--) {
-    res = (res << 6) | (*(++p) & 0x3f);
+    res = (res << 6) | (*(++up) & 0x3f);
   }
   *dst = res;
-  return p + 1;
+  return (const char*)up + 1;
 }
```

Comment:
doing integer math on a char (which can be signed or unsigned depending on the implementation) can be problematic.

Cast char to unsigned char so we can do proper math on it and return it to char later